packages/babel-plugin-remove-graphql-queries/src/index.ts

```diff
@@ -169,34 +236,38 @@ function getGraphQLTag(path) {
   }
 }
 
-function isUseStaticQuery(path) {
+function isUseStaticQuery(path: NodePath<CallExpression>): boolean {
   return (
     (path.node.callee.type === `MemberExpression` &&
       path.node.callee.property.name === `useStaticQuery` &&
-      path.get(`callee`).get(`object`).referencesImport(`gatsby`)) ||
-    (path.node.callee.name === `useStaticQuery` &&
-      path.get(`callee`).referencesImport(`gatsby`))
+      (path.get(`callee`).get(`object`) as NodePath).referencesImport(
+        `gatsby`,
+        ``
+      )) ||
+    ((path.node.callee as Identifier).name === `useStaticQuery` &&
+      path.get(`callee`).referencesImport(`gatsby`, ``))
   )
 }
 
-export default function ({ types: t }) {
+export default function ({ types: t }): PluginObj {
   return {
     visitor: {
-      Program(path, state) {
+      Program(path: NodePath<Program>, state: any): void {
```

Comment:
can we type `state` to avoid `any`? AFAIK we don't pass any options so generic one should work (if there is one)

Reply:
It's hard to tell what `state` is supposed to be because there aren't many resources that I've found on writing babel plugins. Based on [this](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/babel__traverse/index.d.ts) type file, `state` is either the generic type `S` or `any`, and after some digging it looks like `S` is set to `{}` by default. Are you thinking we should just create our own type for `state`?

Reply:
Update on this. I created a custom type for `state` called `IState` and I get an error:

```
Type '(this: {}, path: NodePath<Program>, state: IState) => void' is not assignable to type 'VisitNodeFunction<{}, Program> | VisitNodeObject<{}, Program> | undefined'.
  Type '(this: {}, path: NodePath<Program>, state: IState) => void' is not assignable to type 'VisitNodeFunction<{}, Program>'.
    Types of parameters 'state' and 'state' are incompatible.
      Property 'file' is missing in type '{}' but required in type 'IState'.
```
I think it's because the type of state is expected to be `{}`. However, when I explicitly declare the type like this:
```
Program(path: NodePath<Program>, state: {}): void {
  ...
}
```
I get an error saying `property 'file' does not exist on type '{}'`. I'm unsure how to resolve this issue at the moment. If you know a solution, @pieh, feel free to just add it in.